src/io.c

```diff
@@ -531,6 +500,7 @@ UInt OpenInput (
     // start with an empty line
     STATE(In) = IO()->Input->line;
     STATE(In)[0] = STATE(In)[1] = '\0';
+    STATE(In)++;
```

Comment:
I'm not positive why we used to set the first two elements to 0 here (for look ahead?), so maybe we now need to set 3 elements to 0?

Reply:
Yeah I also was wondering about that. But I couldn't come up with any reason. Note that this code was already exactly like this (well, modulo the `STATE` bit) in GAP 4.4, and that predates the look ahead, which is only used inside the floating point parsing code, which was added in GAP 4.5.

I don't mind clearing three bytes, but I'd prefer to understand *why* that might be necessary?

Reply:
I also couldn't find any sensible reason for this either

Reply:
There are places where we access STATE(In)[1] (and even STATE(In)[2]), but they are only triggered if *STATE(In) is `\`, so they wouldn't be used in this situation, but maybe it was just for safety.

Reply:
Yeah I think that's actually a reasonable explanation: there was/is some code which looks ahead to (1) deal with line continuations and (b) deal with CR/LF vs. CR vs LF. But the latter is mostly done by GetLine2 or code even lower level; and the former (in its current form) is easily enough seen as being correct, as in: it won't access `In[1]` if `In[0]` is 0. And IMHO that's a good invariant to rely on: i.e., the above change is correct not because of some obscure code analysis, but rather because it simply adhere to the basic rule that a C string is terminated by a single null byte. If we have any code in there left which violates that, by expecting a second null byte (and otherwise producing wrong results), then *that* code is buggy and ought to be fixed.

I therefore feel confident to make this change; even if it causes a regression down the line, then this just means it helped us unearth a bug which was covered up before.